addons/binding/org.openhab.binding.networkcamera/src/main/java/org/openhab/binding/networkcamera/internal/ftp/FtpServer.java

```diff
@@ -0,0 +1,536 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.networkcamera.internal.ftp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.ftpserver.FtpServerFactory;
+import org.apache.ftpserver.ftplet.Authentication;
+import org.apache.ftpserver.ftplet.AuthenticationFailedException;
+import org.apache.ftpserver.ftplet.Authority;
+import org.apache.ftpserver.ftplet.AuthorizationRequest;
+import org.apache.ftpserver.ftplet.DefaultFtplet;
+import org.apache.ftpserver.ftplet.FileSystemFactory;
+import org.apache.ftpserver.ftplet.FileSystemView;
+import org.apache.ftpserver.ftplet.FtpException;
+import org.apache.ftpserver.ftplet.FtpFile;
+import org.apache.ftpserver.ftplet.FtpRequest;
+import org.apache.ftpserver.ftplet.FtpSession;
+import org.apache.ftpserver.ftplet.FtpStatistics;
+import org.apache.ftpserver.ftplet.Ftplet;
+import org.apache.ftpserver.ftplet.FtpletContext;
+import org.apache.ftpserver.ftplet.FtpletResult;
+import org.apache.ftpserver.ftplet.User;
+import org.apache.ftpserver.ftplet.UserManager;
+import org.apache.ftpserver.listener.Listener;
+import org.apache.ftpserver.listener.ListenerFactory;
+import org.apache.ftpserver.usermanager.UsernamePasswordAuthentication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple FTP server implementation to receive images from network camera.
+ *
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class FtpServer {
+
+    private Logger logger = LoggerFactory.getLogger(FtpServer.class);
+
+    private int port;
+    int idleTimeout;
+
+    private org.apache.ftpserver.FtpServer server;
+    private static List<FtpServerEventListener> listeners = new ArrayList<FtpServerEventListener>();
+    private HashMap<String, UsernamePassword> authenticationData = new HashMap<String, UsernamePassword>();
+    private MyFTPLet myFTPLet;
+
+    public void startServer(int port, int idleTimeout) {
+        stopServer();
+        this.port = port;
+        this.idleTimeout = idleTimeout;
+        initServer();
+    }
+
+    public void stopServer() {
+        if (server != null) {
+            logger.info("Stopping FTP server");
+            server.stop();
+        }
+    }
+
+    public synchronized void addEventListener(FtpServerEventListener listener) {
+        if (!listeners.contains(listener)) {
+            listeners.add(listener);
+        }
+    }
+
+    public synchronized void addAuthenticationCredentials(String username, String password) throws RuntimeException {
```

Comment:
Please do not throw `RuntimeException`, use a checked exception, or create your own.